panic when trying to lookup against file.directory

lookupFileDirectory.hcl

```hcl
file.directory "test" {
  destination = "/tmp/test"
}

file.content "test-content" {
  destination = "/tmp/test/content1.txt"
  content = "{{lookup `file.directory.test.destination`}}"
}
```

```shell
$ ./converge plan --local samples/lookupFileDirectory.hcl
timestamp="2016-09-21T19:39:33-04:00" level="INFO" msg="serving" addr=":47740" component="rpc"
timestamp="2016-09-21T19:39:33-04:00" level="WARN" msg="skipping module verification" component="client"
panic: reflect: NumField of non-struct type [recovered]
	panic: reflect: NumField of non-struct type

goroutine 33 [running]:
panic(0x5cbf80, 0xc420246b80)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
text/template.errRecover(0xc4200f8e70)
	/usr/local/go/src/text/template/exec.go:146 +0x18e
panic(0x5cbf80, 0xc420246b80)
	/usr/local/go/src/runtime/panic.go:458 +0x243
reflect.(*rtype).NumField(0x622340, 0xc42007d170)
	/usr/local/go/src/reflect/type.go:1003 +0x96
github.com/asteris-llc/converge/render/preprocessor.addFieldsToMap(0xc420267a40, 0x9d2040, 0x622340, 0x622340, 0x0, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/preprocessor/preprocessor.go:288 +0xc7
github.com/asteris-llc/converge/render/preprocessor.addFieldsToMap(0xc420267a40, 0x9d2040, 0x671680, 0xc4200f7de0, 0xc420267a01, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/preprocessor/preprocessor.go:292 +0x194
github.com/asteris-llc/converge/render/preprocessor.fieldMap(0x6abd80, 0x671680, 0x6abd80, 0x671680, 0x2)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/preprocessor/preprocessor.go:280 +0x1f7
github.com/asteris-llc/converge/render/preprocessor.LookupCanonicalFieldName(0x9d2040, 0x671680, 0xc420267999, 0xb, 0xc4200f8030, 0x106fe, 0xc420246b70, 0x10)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/preprocessor/preprocessor.go:313 +0xa0
github.com/asteris-llc/converge/render/preprocessor.HasPath(0x6441c0, 0xc4202674d0, 0xc420246b70, 0x1, 0x1, 0x6bc086, 0x1)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/preprocessor/preprocessor.go:213 +0x191
github.com/asteris-llc/converge/render/preprocessor.EvalTerms(0x6441c0, 0xc4202674d0, 0xc420246b70, 0x1, 0x1, 0xc420267999, 0xb, 0x1, 0x1)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/preprocessor/preprocessor.go:233 +0x74
github.com/asteris-llc/converge/render.(*Renderer).lookup(0xc42026c3c0, 0xc4202449fb, 0x1f, 0x10, 0x10, 0xc4200f83a0, 0x119f5)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/renderer.go:102 +0x36b
github.com/asteris-llc/converge/render.(*Renderer).(github.com/asteris-llc/converge/render.lookup)-fm(0xc4202449fb, 0x1f, 0x0, 0x0, 0x0, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/renderer.go:52 +0x3e
reflect.Value.call(0x5ee7a0, 0xc420246ad0, 0x13, 0x6bc7a9, 0x4, 0xc4202714a0, 0x1, 0x1, 0x9d2040, 0x60e6a0, ...)
	/usr/local/go/src/reflect/value.go:434 +0x5c8
reflect.Value.Call(0x5ee7a0, 0xc420246ad0, 0x13, 0xc4202714a0, 0x1, 0x1, 0x9cd880, 0xc42026c480, 0x5cbf80)
	/usr/local/go/src/reflect/value.go:302 +0xa4
text/template.(*state).evalCall(0xc4200f8df0, 0x5cbf80, 0xc420246ae0, 0x98, 0x5ee7a0, 0xc420246ad0, 0x13, 0x9cd4c0, 0xc4202677d0, 0xc4202449f3, ...)
	/usr/local/go/src/text/template/exec.go:658 +0x530
text/template.(*state).evalFunction(0xc4200f8df0, 0x5cbf80, 0xc420246ae0, 0x98, 0xc420267800, 0x9cd4c0, 0xc4202677d0, 0xc420271480, 0x2, 0x2, ...)
	/usr/local/go/src/text/template/exec.go:530 +0x195
text/template.(*state).evalCommand(0xc4200f8df0, 0x5cbf80, 0xc420246ae0, 0x98, 0xc4202677d0, 0x0, 0x0, 0x0, 0x11008, 0xc42004ebf0, ...)
	/usr/local/go/src/text/template/exec.go:427 +0x6f7
text/template.(*state).evalPipeline(0xc4200f8df0, 0x5cbf80, 0xc420246ae0, 0x98, 0xc4201b50e0, 0xc41ffecc3c, 0xc42004ed30, 0x106fe)
	/usr/local/go/src/text/template/exec.go:400 +0xee
text/template.(*state).walk(0xc4200f8df0, 0x5cbf80, 0xc420246ae0, 0x98, 0x9cd340, 0xc420267830)
	/usr/local/go/src/text/template/exec.go:226 +0x423
text/template.(*state).walk(0xc4200f8df0, 0x5cbf80, 0xc420246ae0, 0x98, 0x9cd6a0, 0xc4202677a0)
	/usr/local/go/src/text/template/exec.go:234 +0x13d
text/template.(*Template).execute(0xc42026c400, 0x9c1dc0, 0xc4201fa620, 0x5cbf80, 0xc420246ae0, 0x0, 0x0)
	/usr/local/go/src/text/template/exec.go:189 +0x204
text/template.(*Template).Execute(0xc42026c400, 0x9c1dc0, 0xc4201fa620, 0x5cbf80, 0xc420246ae0, 0x0, 0xd91b)
	/usr/local/go/src/text/template/exec.go:175 +0x53
github.com/asteris-llc/converge/render/extensions.(*LanguageExtension).Render(0xc4202466a0, 0x5cbf80, 0xc420246ae0, 0x6bf7b5, 0x7, 0xc4202449f1, 0x2c, 0x0, 0x0, 0x0, ...)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/extensions/extensions.go:166 +0x1fc
github.com/asteris-llc/converge/render.(*Renderer).Render(0xc42026c3c0, 0x6bf7b5, 0x7, 0xc4202449f1, 0x2c, 0x10cd4b, 0xc420246700, 0x0, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/renderer.go:53 +0x1f7
github.com/asteris-llc/converge/resource/file/content.(*Preparer).Prepare(0xc4201d9f80, 0x9ce060, 0xc42026c3c0, 0xc42026c3c0, 0x0, 0x0, 0x1)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/resource/file/content/preparer.go:35 +0x69
github.com/asteris-llc/converge/render.pipelineGen.prepareNode(0xc420266c60, 0xc420270160, 0xc4201d94e0, 0x1e, 0xc420266510, 0x5f45c0, 0xc4201d9f80, 0xc42004f320, 0xc420246a80)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/render.go:96 +0x1e0
github.com/asteris-llc/converge/render.(pipelineGen).(github.com/asteris-llc/converge/render.prepareNode)-fm(0x5f45c0, 0xc4201d9f80, 0xc420246a80, 0xc42004f320)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/render.go:67 +0x60
github.com/asteris-llc/converge/executor/either.EitherM.AndThen(0x9c4dc0, 0xc420246a80, 0xc420267650, 0xc42004f3d8, 0xc420271301)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/executor/either/either.go:55 +0x2b2
github.com/asteris-llc/converge/executor.Pipeline.Exec.func1(0x679b60, 0xc420246a90, 0x5e44c0, 0xc420267650, 0x5d4da0, 0xc4202713a0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/executor/pipeline.go:69 +0xcb
github.com/asteris-llc/converge/executor/list.Foldl(0x7198f0, 0x679b60, 0xc420246a90, 0x5d4da0, 0xc420271400, 0xc420246a90, 0x10)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/executor/list/list.go:133 +0xd1
github.com/asteris-llc/converge/executor/list.Foldl(0x7198f0, 0x679b60, 0xc420246a70, 0x5d4da0, 0xc420271460, 0xd91b, 0x679b60)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/executor/list/list.go:133 +0x10a
github.com/asteris-llc/converge/executor.Pipeline.Exec(0x5d4da0, 0xc420271460, 0x679b60, 0xc420246a70, 0xc420246a70, 0x5d4da0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/executor/pipeline.go:71 +0x59
github.com/asteris-llc/converge/render.Render.func1(0xc4201d94e0, 0x1e, 0xc420266c60, 0xc4201d9f80, 0xc42004f701)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/render.go:44 +0x21b
github.com/asteris-llc/converge/graph.transform.func1(0xc4201d94e0, 0x1e, 0x5f45c0, 0xc4201d9f80, 0xc4201d9f80, 0xc42026c380)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/graph/graph.go:554 +0x45
github.com/asteris-llc/converge/graph.rootFirstWalk(0x9cbf00, 0xc4201c3fb0, 0xc420266c60, 0xc420270680, 0x1e, 0xc4201ce000)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/graph/graph.go:423 +0x95c
github.com/asteris-llc/converge/graph.transform(0x9cbf00, 0xc4201c3fb0, 0xc42017bc50, 0x719920, 0xc420270180, 0x6409a0, 0xc4201d9401, 0xc420270180)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/graph/graph.go:555 +0xba
github.com/asteris-llc/converge/graph.(*Graph).RootFirstTransform(0xc42017bc50, 0x9cbf00, 0xc4201c3fb0, 0xc420270180, 0x0, 0x0, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/graph/graph.go:444 +0x55
github.com/asteris-llc/converge/render.Render(0x9cbf00, 0xc4201c3fb0, 0xc42017bc50, 0xc420266510, 0xc420266510, 0xc42017bc50, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/render/render.go:52 +0xdd
github.com/asteris-llc/converge/rpc/pb.(*LoadRequest).Load(0xc4201d8ca0, 0x9cbf00, 0xc4201c3fb0, 0x0, 0x0, 0xc42017d580)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/rpc/pb/loadrequest.go:41 +0x2be
github.com/asteris-llc/converge/rpc.(*executor).Plan(0xc4200862f8, 0xc4201d8ca0, 0x9ce180, 0xc4201c56e0, 0x0, 0xc42016b4c8)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/rpc/executor.go:114 +0x2b7
github.com/asteris-llc/converge/rpc/pb._Executor_Plan_Handler(0x668640, 0xc4200862f8, 0x9cd040, 0xc4200a90e0, 0x38, 0x30)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/rpc/pb/root.pb.go:492 +0x117
github.com/asteris-llc/converge/vendor/google.golang.org/grpc.(*Server).processStreamingRPC(0xc4201c6fa0, 0x9cec60, 0xc4200a9050, 0xc4200e61e0, 0xc4201c3380, 0x9eb840, 0xc4201c3d70, 0x0, 0x0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/google.golang.org/grpc/server.go:686 +0x702
github.com/asteris-llc/converge/vendor/google.golang.org/grpc.(*Server).handleStream(0xc4201c6fa0, 0x9cec60, 0xc4200a9050, 0xc4200e61e0, 0xc4201c3d70)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/google.golang.org/grpc/server.go:770 +0xc36
github.com/asteris-llc/converge/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc4201c5620, 0xc4201c6fa0, 0x9cec60, 0xc4200a9050, 0xc4200e61e0)
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/google.golang.org/grpc/server.go:419 +0xab
created by github.com/asteris-llc/converge/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
	/Users/ryan/Projects/golang/src/github.com/asteris-llc/converge/vendor/google.golang.org/grpc/server.go:420 +0xa3
```

--- a/render/preprocessor/preprocessor.go
+++ b/render/preprocessor/preprocessor.go
@@ -289,8 +289,11 @@ func addFieldsToMap(m map[string]string, t reflect.Type) (map[string]string, err
 		field := t.Field(idx)
 		if field.Anonymous {
 			var err error
-			if m, err = addFieldsToMap(m, interfaceToConcreteType(field.Type)); err != nil {
-				return nil, err
+			anonType := interfaceToConcreteType(field.Type)
+			if anonType.Kind() == reflect.Struct {
+				if m, err = addFieldsToMap(m, anonType); err != nil {
+					return nil, err
+				}
 			}
 			continue
 		}

